### ISSUE TITLE
`--mount` bind-mount creates non-existing directory, contradicting documentation

### ISSUE THREAD

**Issue by Marcono1234** [docker/for-win#11958]
According to the [documentation](https://docs.docker.com/storage/bind-mounts/#differences-between--v-and---mount-behavior) `docker run --mount` with `type=bind` should fail when the directory on the host does not exist:
> If you use `--mount` to bind-mount a file or directory that does not yet exist on the Docker host, Docker does not automatically create it for you, but generates an error.

From what I have heared (have not tested it), this is also the way Docker behaves on Linux.

  - [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: C373B89E-9BCA-4594-B0E3-2A4AC1E72661/20210821232229

### Actual behavior
The non-existing directory on the host is created

### Expected behavior
Docker should fail (as described in the documentation) because the directory does not exist on the host

### Information

  - Windows Version: Windows 10 21H1 (OS Build 19043.1165)
  - Docker Desktop Version: 3.6.0 (67351)
  - WSL2 or Hyper-V backend? WSL2
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: :x:

### Steps to reproduce the behavior
1. Make sure there is no directory called `does-not-exist` in your current directory
2. Run the following Docker command:
    ```
    docker container run --rm --mount type=bind,source="%cd%/does-not-exist",target=/some-dir tianon/true
    ```

**Comment by Milly** [comment]
probably same #10422

**Comment by Marcono1234** [comment]
You are right, it look indeed a lot like #10422; I must have missed that when looking for existing issues. However, that issue is locked, so I would have had to create a new issue either way.

**Comment by brstreetlaw** [comment]
I'm hesitant to create a new issue but here's what I was experiencing, and the workaround I finally discovered. Can someone suggest whether it is a new issue or not?

### Expected behaviour
Bind mounts (whether in **docker run**, **docker-compose**, or even using **portainer.io**) should be able to specify a directory within the host Windows filesystem, or in the very least, how that path would appear within the WSL filesystem (e.g. **/mnt/c/Users/Public/Downloads**:/downloads)

### Actual behaviour
Bind mounts in the format above end up pointing to an obscure directory that Windows sees as created within **\\wsl.localhost\docker-desktop-data\mnt\wsl\docker-desktop-data\tarcache\entries\docker.tar\(some_long_hex_id}\containers\services\docker\rootfs\mnt\c\users\public\downloads\** (in the example above)

### Information
Windows Version: Windows 11 21H2 (OS Build 22000.376)
Docker Desktop Version: 4.3.1
WSL2 or Hyper-V backend? WSL2
Are you running inside a virtualized Windows e.g. on a cloud server or a VM: No

### Workaround

1. Don't use PowerShell to deploy containers. Use Ubuntu bash shell within WSL to execute **docker run** and **docker-compose**; bind mounts to the Ubuntu instance of WSL work as expected, and so do bind mounts to the host Windows filesystem using **/mnt/c/whatever**
2. When deploying with portainer.io, first install and use a web browser via WSLg that runs within the WSL VM (e.g. open Windows Terminal, start an Ubuntu bash shell, **cd /tmp** and run **wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb** followed by **sudo dpkg -i google-chrome-stable_current_amd64.deb; sudo apt-get -f install**

...then launch _Google Chrome (Ubuntu)_ from Windows start menu, and point that browser to **https://localhost:9443/**

...then containers deployed via the portainer.io interface can bind mount to directories in the Ubuntu and/or the host Windows filesystem using any path that can be browsed via WSL Ubuntu bash shell.

Later on, inspecting the configuration of those deployed containers, the bind mounts will appear to have been changed to host paths of **/run/desktop/mnt/host/wsl/docker-desktop-bind-mounts/Ubuntu/(some_long_hex_id)** but when you browse **\\wsl.localhost\docker-desktop-data\mnt\wsl\docker-desktop-bind-mounts\Ubuntu** in Windows, those long hex directories are inaccessible and show 0 bytes as their size - making me think they are symlinks.

Docker Desktop can now start and stop these containers, access CLI, etc. from the Windows host.

This is the **only** way that I have been able to make containers share directories and files with the Windows host, or WSL Ubuntu for that matter.

**Comment by Marcono1234** [comment]
@brstreetlaw, to me that sounds like a different issue (though the maintainers might think differently about this). My report was only about the fact that the documentation contradicts the actual behavior regarding whether non-existent directories are created. However, for me the directories were created in the correct location.

**Comment by sarisia** [comment]
Thank you for raising this issue again (I'm the author of #10422).

I've confirmed this is still the case with Docker Desktop.

My workaround is remove Docker Desktop and install docker-ce. It works very well for me for months
and still haven't have any critical issues.

Just install docker-ce [as described in the official docs](https://docs.docker.com/engine/install/ubuntu/)
and then append the lines below to `/etc/wsl.conf` ([ref](https://docs.microsoft.com/en-us/windows/wsl/wsl-config#boot-settings)):

```
[boot]
command = service docker start
```

I feel installing docker-ce manually is a bit of a pain so hope this issue resolves...

**Comment by Marcono1234** [comment]
Still the case for Docker Desktop 4.14.1 (91661), Docker version 20.10.21, build baeda1f

**Comment by Marcono1234** [comment]
/remove-lifecycle stale

@docker-desktop-robot seems pretty pointless to me. It only causes a lot of noise and closes valid issues. And apparently anyone can disable it with `/lifecycle frozen`. So why does this bot exist in the first place?

**Comment by Gibbz** [comment]
This is still happening on Linux.
When can we expect to see a fix?

**Comment by Marcono1234** [comment]
> This is still happening on Linux.

@Gibbz, this report here is about the behavior of Docker Desktop for Windows, and also this repository `docker/for-win` here is only for issues about Docker Desktop for Windows. If you experience the same issue for Docker Desktop for Linux, you should probably report it at https://github.com/docker/desktop-linux. Please link to this issue here then.
